### ISSUE TITLE
Can't play encrypted DVD's

### ISSUE THREAD

**Issue by DavidMcKenna** [komh/vlc-os2#13]
With version 3.0.2 of VLC, I can not play any encrypted DVD. When I try, the message console shows this error:

DVD could not read - 1/4 blocks at 0x01

over and over, and the disc player makes a clicking noise as if searching for something. Non-encrypted DVD's play fine.

**Comment by komh** [comment]
Any other version worked ? How about MPlayer ?

Unfortunately, I cannot reproduce this problem here.

And attach vlc-log.txt and vlc-err.txt as described in README, please.

Thanks.

**Comment by DavidMcKenna** [comment]
[vlc-err.txt](https://github.com/komh/vlc-os2/files/2026261/vlc-err.txt)
[vlc-log.txt](https://github.com/komh/vlc-os2/files/2026262/vlc-log.txt)

Logs uploaded. The last version I tried I think was 2.2.4 and it worked, but it was about 4 months ago on a different computer. BTW, I am using UNIAUD and Panorama. I'll have to get a copy of MPlayer to test.

**Comment by komh** [comment]
In your log, I've found that dvdnav module failed because it could not read from DVD. But I'm not sure what it is the problem. That is, it may be due to DVD titles themselves or dvdread/dvdnav module of VLC.

I want to know that MPlayer for OS/2 can play them or VLC on other platforms such as Windows can.

**Comment by DavidMcKenna** [comment]
I installed Mplayer and SMPlayer for OS/2 on my desktop machine and I also can't play encrypted discs with them. I don't have Windows on that machine so can't test VLC that way.

I do have ArcaOS and Windows on my laptop, so I will move to testing there. I get the exact same problem and errors with encrypted DVD's on my laptop, but unencrypted DVD's play fine. On Windows with VLC 3.0.2 on the laptop, ALL DVD's I tried play fine, so it seems the hardware works. I should point out that on both OS/2 and Windows I tested with default settings. Maybe I need to change some settings on ArcaOS?

 One other thing happened: I was playing the unencrypted DVD (The Third Man - Criterion Collection version) and I hit the 'Stop' button on the taskbar at the bottom of the screen to stop playing. I got a loud beep and VLC locked up for about 10 seconds, then disappeared. After that the WPS was unstable and I had to reboot. After reboot, tried it again and the same thing happened. It produced a *.trp file I will upload.

**Comment by DavidMcKenna** [comment]
file:///C:/usr/local/bin/0047_03.TRP

**Comment by komh** [comment]
Check the region code of your DVD drive. The region code of your DVD drive should match with the region code of your DVD title. Try

http://www.os2site.com/sw/mmedia/video/regionset.zip

or

[regionset.zip](https://github.com/komh/vlc-os2/files/2048032/regionset.zip)

And I could not view your .TRP file. Upload again properly, please...

**Comment by DavidMcKenna** [comment]
Sorry for long absence, no time to work with this until now.
 I tried regionset with one of the unplayable DVD's and got this:

[C:\]regionset s:
Current Region Code settings:
RPC Phase: II
type: NONE
vendor resets available: 4
user controlled changes resets available: 5
drive plays discs from region(s):, mask=0xFF

Would you like to change the region setting of your drive? [y/n]:n
-------------------------------------------------------------------------------------

 Not sure what it means. I'll also try to reproduce the trap and try uploading again (*.trp is not one of the file types listed when I click on 'selecting item' at the bottom of the comment box).

**Comment by komh** [comment]
The region code of Your DVD drvie has not been set. Please, set the region code according to your region. See https://en.wikipedia.org/wiki/DVD_region_code#Region_codes_and_countries for reference.

And for crash, it seems to be in Qt. Anyway tell me the value of your QT_PM_NO_DIVE and QT_PM_DIVE env var. In addition, please test kva modes of VLC, which are VMAN and DIVE. You can find the settings in Tools - Preferences - Video - KVA - Video Mode.

**Comment by DavidMcKenna** [comment]
OK, regionset now shows:

[C:\]regionset s:
Current Region Code settings:
RPC Phase: II
type: SET
vendor resets available: 4
user controlled changes resets available: 4
drive plays discs from region(s): 1, mask=0xFE

Would you like to change the region setting of your drive? [y/n]:n

 but it didn't help, still can't play any encrypted DVD's (same error). I don't have any QT_ settings in my environment (so I assume all are default) but I tried explicitly setting QT_PM_NO_DIVE (I'm using panorama) but it didn't help. Also tried different video modes but again nothing helped. Downloaded the new VLC 3.0.3 and no help there either....

**Comment by komh** [comment]
Maybe do you have dvdcss2.dll in a directory listed in LIBPATH ? If so, I'm sorry that I have no idea about this because I cannot reproduce this here.

For crash, maybe could you try other video driver than Panorama ?

**Comment by DavidMcKenna** [comment]
No, only 1 dvdcss2.dll on my system(s):

[C:\]dir dvdcss2.dll /s

The volume label in drive C is ARCAOS.
The Volume Serial Number is 593B:75BB.

Directory of C:\usr\local\lib

 1-17-18 10:34p        24,936    394 a---  dvdcss2.dll
        1 file(s)      24,936 bytes used

  I can try installing Snap (both my laptop and desktop computers are too new for accelerated drivers) using the fallback driver over the weekend - I'll report back on that.

 One possible problem - both my desktop and laptop report a strange error when I enter on the command line: 'Copy OS2AHCI$ ahci.log'. In the log I see many lines like:

port #1 interrupt error status: 0x40000001; restarting port

 The port listed is the one my DVD drive is on (both laptop and desktop). Not sure why this happens - maybe it is why VLC doesn't work (but strange that non-encrypted discs DO work). Are you using AHCI and do you see this message?

**Comment by komh** [comment]
Sorry I don't use OS2AHCI.

**Comment by DavidMcKenna** [comment]
Finally got around to installing SNAP (using the VBE driver). It makes no difference, I still get the 'DVD could not read - 1/4 blocks at 0x01' error when I try to play encrypted DVD's, but non-encrypted DVD's play fine. This is pointing more and more at the AHCI driver.... I know there is an update to it waiting for release testing. Hope it helps.... I'll report when it is released.

**Comment by komh** [comment]
Ah, I suggested SNAP to test crash. Still crash even on SNAP ?

Thanks for your testing!

**Comment by DavidMcKenna** [comment]
Right... sorry I forgot. I can stop playing non-encrypted DVD's no problems when using SNAP (in VBE mode). Think I will keep using SNAP on my laptop for now - the performance seems slightly worse than Panorama, but I can play Youtube videos and DVD's without dropped frames. Wonder why Panorama crashes?...

**Comment by komh** [comment]
Ok, good to know that. It's known that Panorama has some bugs. I think, this is the case, too. And it seems that this crash is in Qt4 not VLC itself.

Thanks for your test result.

**Comment by DavidMcKenna** [comment]
Hi,

 It's been a long time, and I still cannot play encrypted DVD's on any bare metal OS/2 installation I have, even with your latest 3.0.23 version. However, I have recently started running ArcaOS in a VirtualBox VM and I CAN play them there - using the IDE hardware emulation with the DANIS driver. If I switch to AHCI, then I cannot play them and I get the error already described.

 So it seems there is a limitation with the OS2AHCI driver. I'll report this to ArcaOS and see what they say...

**Comment by komh** [comment]
Oh, good to know that.

Additionally, maybe do you have daniatapi.flt or ibmatapi.flt with OS2AHCI driver ?

**Comment by DavidMcKenna** [comment]
Yes, I have daniatap.flt in my CONFIG.SYS on both bare metal and VirtualBox machines with the OS2AHCI driver.

 One thing that is new: I usually tried to run the DVD from 'Open Disc' and got the error mentioned, but if I run 'Open Directory' and point to the disc it tries to run, but hangs the VLC app and I have to kill it. On one occasion (on the VirtualBox machine) it actually tried to play a very pixelated intro, then crashed.

 I have entered a ticket for this at ArcaNoae. Let's see what they say...

**Comment by DavidMcKenna** [comment]
FYI - it's been over a month and ArcaNoae has not responded to my ticket. Not sure why...
